app/src/main/java/com/example/sharewhatyoucanproject/donor/AddDonorViewModel.kt

```diff
@@ -17,8 +18,8 @@ import kotlin.collections.HashMap
 class AddDonorViewModel(
     private val db: FirebaseFirestore,
     private val storageReference: StorageReference,
-
-    ) : ViewModel() {
+    private var auth: FirebaseAuth = FirebaseAuth.getInstance(),
```

Comment:
Default to Val instead of var if the variable is immutable

Reply:
done